src/backgroundblurprocessor/BackgroundBlurProcessorBuiltIn.ts

```diff
@@ -0,0 +1,139 @@
+// SPDX-License-Identifier: Apache-2.0
+
+import BackgroundBlurOptions from './BackgroundBlurOptions';
+import BackgroundBlurProcessorProvided from './BackgroundBlurProcessorProvided';
+import { BlurStrengthMapper } from './BackgroundBlurStrength';
+import BackgroundFilterSpec from './BackgroundFilterSpec';
+
+/**
+ * The [[BackgroundBlurProcessorBuiltIn]] uses the browser's built-in capability to apply blurring to
+ * the background image as apposed to [[BackgroundBlurProcessorProvided]] that uses WASM and
+ * TensorFlow Lite to apply the blur.
+ */
+
+/** @internal */
+export default class BackgroundBlurProcessorBuiltIn extends BackgroundBlurProcessorProvided {
+  private blurredImage: ImageData;
+  private blurCanvas: HTMLCanvasElement = document.createElement('canvas') as HTMLCanvasElement;
+  private blurCanvasCtx = this.blurCanvas.getContext('2d');
+
+  /**
+   * A constructor that will apply default values if spec and strength are not provided.
+   * If no spec is provided the selfie segmentation model is used with default paths to CDN for the
+   * worker and wasm files used to process each frame.
+   * @param spec The spec defines the assets that will be used for adding background blur to a frame
+   * @param options How much blur to apply to a frame
+   */
+  constructor(spec?: BackgroundFilterSpec, options?: BackgroundBlurOptions) {
+    super(spec, options);
+
+    this.blurCanvas.width = this.spec.model.input.width;
+    this.blurCanvas.height = this.spec.model.input.height;
+    this.logger.info('BackgroundBlur processor using builtin blur');
+  }
+
+  drawImageWithMask(inputCanvas: HTMLCanvasElement, mask: ImageData): void {
+    // Mask will not be set until the worker has completed handling the predict event. Until the first frame is processed,
+    // the whole frame will be blurred.
+    const blurredImage = this.blurredImage;
+    const { canvasCtx, targetCanvas } = this;
+    const { width, height } = targetCanvas;
+
+    if (!mask || !blurredImage) {
+      canvasCtx.clearRect(0, 0, width, height);
+      return;
+    }
+
+    const scaledCtx = this.scaledCanvas.getContext('2d');
+
+    scaledCtx.putImageData(mask, 0, 0);
+    this.blurCanvasCtx.putImageData(this.blurredImage, 0, 0);
+
+    // draw the mask
+    canvasCtx.save();
+    canvasCtx.clearRect(0, 0, width, height);
+    canvasCtx.drawImage(this.scaledCanvas, 0, 0, width, height);
+
+    // Only overwrite existing pixels.
+    canvasCtx.globalCompositeOperation = 'source-in';
+    // draw image over mask...
+    canvasCtx.drawImage(inputCanvas, 0, 0, width, height);
+
+    // draw under person
+    canvasCtx.globalCompositeOperation = 'destination-over';
+    canvasCtx.drawImage(this.blurCanvas, 0, 0, width, height);
+    canvasCtx.restore();
+  }
+
+  /**
+   * This setter will allow setting the blur strength mid stream. The process method will use this
+   * setting on the next frame that is processed.
+   */
+  setBlurStrength(blurStrength: number): void {
```

Comment:
Can we drop this documentation as it extends `BackgroundBlurProcessorProvided` which implements `BackgroundBlurProcessor` and the `setBlurStrength` I think already has the same documentation?

Reply:
updated